tfhe/src/core_crypto/algorithms/test/noise_distribution/lwe_encryption_noise.rs

```diff
@@ -0,0 +1,166 @@
+use super::*;
+use crate::core_crypto::commons::test_tools::{torus_modular_distance, variance};
+
+fn lwe_encrypt_decrypt_noise_distribution_custom_mod<Scalar: UnsignedTorus + CastInto<usize>>(
+    params: TestParams<Scalar>,
+) {
+    let lwe_dimension = params.lwe_dimension;
+    let lwe_modular_std_dev = params.lwe_modular_std_dev;
+    let ciphertext_modulus = params.ciphertext_modulus;
+    let message_modulus_log = params.message_modulus_log;
+    let encoding_with_padding = get_encoding_with_padding(ciphertext_modulus);
+    let expected_variance = Variance(lwe_modular_std_dev.get_variance());
+
+    let mut rsc = TestResources::new();
+
+    const NB_TESTS: usize = 1000;
+    let msg_modulus = Scalar::ONE.shl(message_modulus_log.0);
+    let mut msg = msg_modulus;
+    let delta: Scalar = encoding_with_padding / msg_modulus;
+
+    let num_samples = NB_TESTS * <Scalar as CastInto<usize>>::cast_into(msg);
+    let mut noise_samples = Vec::with_capacity(num_samples);
+
+    while msg != Scalar::ZERO {
+        msg = msg.wrapping_sub(Scalar::ONE);
+        for _ in 0..NB_TESTS {
+            let lwe_sk = allocate_and_generate_new_binary_lwe_secret_key(
+                lwe_dimension,
+                &mut rsc.secret_random_generator,
+            );
+
+            let mut ct = LweCiphertext::new(
+                Scalar::ZERO,
+                lwe_dimension.to_lwe_size(),
+                ciphertext_modulus,
+            );
+
+            let plaintext = Plaintext(msg * delta);
+
+            encrypt_lwe_ciphertext(
+                &lwe_sk,
+                &mut ct,
+                plaintext,
+                lwe_modular_std_dev,
+                &mut rsc.encryption_random_generator,
+            );
+
+            assert!(check_content_respects_mod(&ct, ciphertext_modulus));
+
+            let decrypted = decrypt_lwe_ciphertext(&lwe_sk, &ct);
+
+            let decoded = round_decode(decrypted.0, delta) % msg_modulus;
+
+            assert_eq!(msg, decoded);
+
+            let torus_distance =
+                torus_modular_distance(plaintext.0, decrypted.0, ciphertext_modulus);
+            noise_samples.push(torus_distance);
+        }
+    }
+
+    let measured_variance = variance(&noise_samples);
+    assert!((expected_variance.0 - measured_variance.0).abs() < 0.05 * expected_variance.0);
```

Comment:
Could it be interesting to use a power of 2 instead of 0.05?

Reply:
yes, would avoid some rounding here, I'll udpate